intake_elasticsearch/elasticsearch_table.py

```diff
@@ -76,13 +84,23 @@ def _get_partition(self, _):
         https://stackoverflow.com/questions/41655913/elk-how-do-i-retrieve-more-than-10000-results-events-in-elastic-search
         """
         import pandas as pd
+        slice_id = None
+        slice_max = None
+        if isinstance(_, tuple):
+            slice_id, slice_max = _
         if self._dataframe is None or self.part:
-            results = self._run_query()
+            results = self._run_query(slice_id=slice_id, slice_max=slice_max)
             df = pd.DataFrame([r['_source'] for r in results['hits']['hits']])
             self._dataframe = df
```

Comment:
Should not reset `self._dataframe` any more, I don't think, it should just remain the one derived from the first few rows of the dataset. The the following lines can look like
```
if df.empty:
    df = self._dataframe[:0]

..
return df
```

I am no longer sure what `self.part`'s purpose perhaps it's no longer needed.